tests/registration_tests.c

```diff
@@ -17,12 +17,75 @@
  *******************************************************************************/
 
 #include "CUnit/CUnit.h"
+#include "connection.h"
+#include "er-coap-13/er-coap-13.h"
 #include "internals.h"
 #include "liblwm2m.h"
 #include "tests.h"
 
+#include <string.h>
+
 #ifdef LWM2M_SERVER_MODE
 
+static void create_test_registration_message(coap_packet_t *packet) {
```

Comment:
This function sounds very generic but creates a very specific message for the test below. Why not having these lines directly in the test function? If we add more tests we would need a new function anyway.

Reply:
My idea is to have, in the future, multiple helper functions that provide some testing messages for different tests.
Having this in a separate function gives it a descriptive name and make the test function a bit smaller.